util/topgen/templates/plic_all_irqs_test.c.tpl

```diff
@@ -98,45 +110,70 @@ void ottf_external_isr(uint32_t *exc_info) {
 
   top_${top["name"]}_plic_peripheral_t peripheral = (top_${top["name"]}_plic_peripheral_t)
       top_${top["name"]}_plic_interrupt_for_peripheral[plic_irq_id];
-  CHECK(peripheral == peripheral_expected,
-        "Interrupt from incorrect peripheral: exp = %d, obs = %d",
-        peripheral_expected, peripheral);
+  if (allow_csrng_irq && kBootStage == kBootStageOwner &&
+      peripheral != peripheral_expected &&
+      peripheral == kTopEarlgreyPlicPeripheralCsrng) {
```

Comment:
Should we refer to #20747 at this point?

Reply:
I added references at other points. A reference here will not hurt. Done.